src/util.ts

```diff
@@ -371,7 +371,13 @@ function resolveToolsGopath(): string {
 }
 
 export function getBinPath(tool: string): string {
-	return getBinPathWithPreferredGopath(tool, (tool === 'go') ? [] : [getToolsGopath(), getCurrentGoPath()], vscode.workspace.getConfiguration('go', null).get('alternateTools'));
+	const alternateTools: { [key: string]: string; } = vscode.workspace.getConfiguration('go', null).get('alternateTools');
+
+	return getBinPathWithPreferredGopath(
+		tool,
+		(tool === 'go') ? [] : [getToolsGopath(), getCurrentGoPath()],
+		resolveObjectPaths(alternateTools),
```

Comment:
`getBinPathWithPreferredGopath` only needs to deal with the current tool.
So, instead of resolving the path for all the alternate tools, why not resolve the path for the given tool and just pass that to `getBinPathWithPreferredGopath` ?

i.e update `getBinPathWithPreferredGopath` to take `alternateToolPath` (a string) instead of `alternateTools` (a key-value map)

Reply:
That does make more sense, I'll change it now. I was focussing more on trying to make the build pass 😄.